spark/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryGetResultSpecTest.java

```diff
@@ -0,0 +1,322 @@
+              interaction.emrJobUpdateJobState(JobRunState.SUCCESS);
+              return result;
+            })
+        .assertQueryResults("running", null)
+        .withoutInteraction()
+        .assertQueryResults("SUCCESS", ImmutableList.of());
+  }
+
+  @Test
+  public void testStreamingQueryGetResult() {
+    // Create mock index with index state refreshing
+    mockIndex.createIndex();
+    mockIndexState.refreshing();
+    try {
+      createAsyncQuery(
+              "CREATE SKIPPING INDEX ON mys3.default.http_logs "
+                  + "(l_orderkey VALUE_SET) WITH (auto_refresh = true)")
+          .withInteraction(
+              interaction -> {
+                interaction.emrJobWriteResultDoc(createEmptyResultDoc(interaction.queryId));
+                interaction.emrJobUpdateJobState(JobRunState.SUCCESS);
+                return interaction.pluginSearchQueryResult();
+              })
+          .assertQueryResults("SUCCESS", ImmutableList.of());
+    } finally {
+      mockIndex.deleteIndex();
+      mockIndexState.deleted();
+    }
+  }
+
+  @Test
+  public void testStreamingQueryGetResultWithConcurrentEmrJobUpdate() {
+    // Create mock index with index state refreshing
+    mockIndex.createIndex();
+    mockIndexState.refreshing();
+    try {
+      createAsyncQuery(
+              "CREATE SKIPPING INDEX ON mys3.default.http_logs "
+                  + "(l_orderkey VALUE_SET) WITH (auto_refresh = true)")
+          .withInteraction(
+              interaction -> {
+                JSONObject result = interaction.pluginSearchQueryResult();
+                interaction.emrJobWriteResultDoc(createEmptyResultDoc(interaction.queryId));
+                interaction.emrJobUpdateJobState(JobRunState.SUCCESS);
+                return result;
+              })
+          .assertQueryResults("running", null)
+          .withoutInteraction()
+          .assertQueryResults("SUCCESS", ImmutableList.of());
+    } finally {
+      mockIndex.deleteIndex();
+      mockIndexState.deleted();
+    }
+  }
+
+  @Test
+  public void testDropIndexQueryGetResult() {
+    // Create mock index with index state refreshing
+    mockIndex.createIndex();
+    mockIndexState.refreshing();
+
+    LocalEMRSClient emrClient = new LocalEMRSClient();
+    emrClient.setJobState("Cancelled");
+    createAsyncQuery(mockIndex.query, emrClient)
+        .withoutInteraction()
+        .assertQueryResults("SUCCESS", ImmutableList.of());
+  }
+
+  @Test
+  public void testDropIndexQueryGetResultWithResultDocRefreshDelay() {
+    // Create mock index with index state refreshing
+    mockIndex.createIndex();
+    mockIndexState.refreshing();
+
+    LocalEMRSClient emrClient = new LocalEMRSClient();
+    emrClient.setJobState("Cancelled");
+    createAsyncQuery(mockIndex.query, emrClient)
+        .withInteraction(interaction -> new JSONObject()) // simulate result index refresh delay
+        .assertQueryResults("running", null)
+        .withoutInteraction()
```

Comment:
withoutInteraction() naming is not clear. could we have better name?

Reply:
Sure, removed and inlined the single-line for clarify in https://github.com/opensearch-project/sql/pull/2443/commits/0dc8b4d94ecaef6c5bb05259c85196359584440c. Thanks!